docs/features/pixi_global_manifest.md

```diff
@@ -0,0 +1,222 @@
+# Pixi Global Manifest
+
+## Motivation
+
+`pixi global` is currently limited to imperatively managing CLI packages.
+The next iteration of this feature should fulfill the following needs:
+
+- Shareable global environments.
+- Managing complex environments with multiple packages as dependencies
+- Flexible exposure of binaries
+
+## Design Considerations
+
+There are a few things we wanted to keep in mind in the design:
+
+1. **User-friendliness**: Pixi is a user focused tool that goes beyond developers. The feature should have good error reporting and helpful documentation from the start.
+2. **Keep it simple**: The CLI should be all you strictly need to interact with global environments.
+3. **Unsurprising**: Simple commands should behave similar to traditional package managers.
+4. **Human Readable**: Any file created by this feature should be human-readable and modifiable.
+
+## Manifest
+
+```toml title="pixi-global.toml"
+# The name of the environment is `python`
+# It will expose python, python3 and python3.11, but not pip
+[envs.python.dependencies]
+python = { spec = "3.11.*", expose_binaries = "auto" }
+pip = "*"
+
+# The name of the environment is `python_3_10`
+# It will expose python3.10
+[envs.python_3_10.dependencies]
+python = { spec = "3.10.*", expose_binaries = { "python3.10"="python" } }
```

Comment:
It was a point of discussion. I don't have strong feelings either way.

The point we made below was that you could "auto" expose one package in the environment, but not expose another package in the environment. It could be a bit surprising if you inject `hypercorn` and suddenly have a new binary in your `PATH`.

Do you think that is either not that important or is too annoying/hard to implement anyway?

Reply:
I think the tool should always update the list of exposed binaries in the toml file, but map those that are not exposed as

```
foo = null
```

So that the user knows which binaries are available to "expose" in the first place. 

Maybe to make it nicer in TOML we should not add them to the package section but to it's own section like so:

```toml
[envs.python_310.exposed_binaries]
python = "python"
"python@3.10" = "python3.10"
python3 = null
```

If you are in a situation like the following:

```
[envs.python_310.exposed_binaries]
python = "python"
# and nothing else
```

then pixi should, upon running `sync`

inject:

```
[envs.python_310.exposed_binaries]
python = "python"
"python3.10" = null
python3 = null
```

Reply:
> The point we made below was that you could "auto" expose one package in the environment, but not expose another package in the environment.

My point was more that this is not tied to a package perse but more to an environment. Binaries are placed by all sorts of packages in an environment. To me it makes more sense to add this like in wolfs example:

```toml
[envs.python_310.exposed_binaries] 
python = "python"
```

It doesnt matter which package installs the binary.

Reply:
I am happy with both setting the `exposed_binary` on the environment level and mentioning all exposed binaries explicitly.

I will have a good look, how the manifest and other commands fit with that and report back.

Reply:
@wolfv TOML doesn't seem to have the concept of `null`: https://github.com/toml-lang/toml/issues/30

Do you still like your example, when the values that are `null` are instead not mentioned at all?